File: src/amount.py
from const import TransactionType

class Amount:

    def __init__(self, db):
        self.db = db


    def calculate(self, ticker):
        table = self.db.table(ticker)

        amount = 0
        for row in table:
            if row["type"] == TransactionType.BUY.name:
                amount += row["amount"]
            elif row["type"] == TransactionType.SELL.name:
                amount -= row["amount"]
            else:
                raise Exception("Transaction type [" + row["type"] + "] not implemented") 

        return amount

File: src/report/amount.rs
use rental_rod::db::table::Table;
use rust_decimal::Decimal;

use crate::{util::{error::CryptoError, helper::Transaction}, transaction::transaction_type::TransactionType};

pub fn get_amount(table: &mut Table) -> Result<Decimal, CryptoError> {
    let mut total = Decimal::ZERO;

    for line in table.get_lines() {
        let t_type = line.get_type()?;
        let amount = line.get_amount()?;

        if t_type == TransactionType::Buy.value() {
            total = total + amount;
        }
        else if t_type == TransactionType::Sell.value() {
            total = total - amount;
        }
        else {
            return Err(CryptoError::Custom(format!("Type [{}] is not supported", &t_type)));
        }
    }

    Ok(total)
}